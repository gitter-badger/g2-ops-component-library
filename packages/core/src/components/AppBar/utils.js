import { compose } from 'ramda'

const FIND_ALL_UNDERSCORES = /_+/g
const FIND_FIRST_LETTER_REGEX = /\w\S*/g

export const replaceUnderscoreWithSpace = (string) =>
  string && string.length > 0 && string.replace(FIND_ALL_UNDERSCORES, ' ')

export const capitalizeFirstLetter = (string) =>
  string &&
  string.replace(
    FIND_FIRST_LETTER_REGEX,
    (match) => match && `${match.charAt(0).toUpperCase()}${match.substr(1).toLowerCase()}`
  )

export const beautifyRoleText = compose(capitalizeFirstLetter, replaceUnderscoreWithSpace)