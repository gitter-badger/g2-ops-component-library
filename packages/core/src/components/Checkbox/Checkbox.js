// @flow

import type { Node } from 'react'
import React from 'react'

import { Checkbox as FabricCheckbox } from 'office-ui-fabric-react/lib/Checkbox'

type CheckboxPropTypes = {
  /** label to be displayed */
  label: string,
  /** isChecked to indicate checked state */
  isChecked?: boolean,
  /** Custom render function for rendering Label */
  labelRenderer?: (any, (any) => Node) => Node,
  /** onChange event handler */
  handleChange?: (SyntheticMouseEvent<HTMLInputElement>, boolean) => void,
  /** Custom styles for Checkbox */
  styles?: { [string]: mixed },
  /** extra props */
  otherProps?: any,
}

export const Checkbox = ({
  label,
	isChecked,
	checked,
	defaultChecked,
  handleChange,
  labelRenderer,
  styles,
  ...otherProps
}: CheckboxPropTypes) => (
  <FabricCheckbox
    label={label}
    checked={isChecked || checked || defaultChecked}
    onChange={handleChange}
    styles={styles}
    onRenderLabel={labelRenderer}
    {...otherProps}
  />
)
