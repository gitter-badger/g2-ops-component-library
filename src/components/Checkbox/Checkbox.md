Checkbox Example:

```js
<Checkbox 
  label={'All Facilities'}
/>
```