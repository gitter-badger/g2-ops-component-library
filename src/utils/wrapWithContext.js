import React from 'react'
import WithContext from 'react-with-context'
import baseTheme from 'material-ui/styles/baseThemes/lightBaseTheme'
import createMuiTheme from 'material-ui/styles/getMuiTheme'

const wrapWithContext = (component, context) => <WithContext context={context}>{component}</WithContext>

export const theme = createMuiTheme(baseTheme)

export const wrapWithMaterialUIContext = (component) =>
  wrapWithContext(component, { muiTheme: createMuiTheme(baseTheme) })

export default wrapWithContext
