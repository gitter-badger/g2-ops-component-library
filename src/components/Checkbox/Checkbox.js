import React from 'react'
import PropTypes from 'prop-types'
import { Checkbox as FabricCheckbox } from 'office-ui-fabric-react/lib/Checkbox'

const checkBoxPropTypes = {
  label: PropTypes.string.isRequired,
  isChecked: PropTypes.bool,
  /** Custom render function for rendering Label */
  onRenderLabel: PropTypes.func,
  handleChange: PropTypes.func,
  styles: PropTypes.object
}

const Checkbox = ({ label, isChecked, handleChange, onRenderLabel, styles, ...otherProps }) => (
  <FabricCheckbox
    label={label}
    checked={isChecked}
    onChange={handleChange}
    styles={styles}
    onRenderLabel={onRenderLabel}
    {...otherProps}
  />
)

Checkbox.propTypes = checkBoxPropTypes

export default Checkbox