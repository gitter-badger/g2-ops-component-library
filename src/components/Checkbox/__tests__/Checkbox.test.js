import React from 'react'
import renderer from 'react-test-renderer'

import { wrapWithMaterialUIContext } from 'utils/wrapWithContext'

import Checkbox from '../Checkbox'

describe('<Checkbox />', () => {
  test('should render properly', () => {
    const tree = renderer.create(wrapWithMaterialUIContext(<Checkbox label="Custom Label" />))
    expect(tree).toMatchSnapshot()
  })
})
