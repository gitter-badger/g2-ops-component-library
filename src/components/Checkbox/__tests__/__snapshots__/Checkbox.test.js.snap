// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> should render properly 1`] = `
<button
  aria-checked={false}
  aria-describedby={undefined}
  aria-disabled={undefined}
  aria-label={undefined}
  aria-labelledby={undefined}
  className="ms-Checkbox is-enabled css-32"
  disabled={undefined}
  id="checkbox-0"
  name={undefined}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  role="checkbox"
  type="button"
>
  <label
    className="ms-Checkbox-label css-33"
    htmlFor="checkbox-0"
  >
    <div
      className="ms-Checkbox-checkbox css-34"
    >
      <i
        aria-hidden={true}
        aria-label={undefined}
        className="root-37 css-35"
        data-icon-name="CheckMark"
        role="presentation"
      />
    </div>
    <span
      className="ms-Checkbox-text css-36"
    >
      Custom Label
    </span>
  </label>
</button>
`;
